lib/cisco_node_utils/bridge_domain.rb

```diff
@@ -0,0 +1,247 @@
+      return hash if bd_list.nil?
+
+      bd_list.each do |id|
+        hash[id] = BridgeDomain.new(id, false)
+      end
+      hash
+    end
+
+    ########################################################
+    #                      PROPERTIES                      #
+    ########################################################
+
+    # Bridge-Domain Shutdown case
+    def shutdown
+      result = config_get('bridge_domain', 'shutdown', @bd_ids)
+      # Valid result is either: "active"(aka no shutdown) or "shutdown"
+      result[/DOWN/] ? true : false
+    end
+
+    def shutdown=(val)
+      no_cmd = (val) ? '' : 'no'
+      result = config_set('bridge_domain', 'shutdown', @bd_ids, no_cmd)
+      cli_error_check(result)
+    rescue CliError => e
+      raise "[bridge-domain #{@bd_ids}] '#{e.command}' : #{e.clierror}"
+    end
+
+    def default_shutdown
+      config_get_default('bridge_domain', 'shutdown')
+    end
+
+    # Bridge-Domain name assigning case
+    def name
+      config_get('bridge_domain', 'name', @bd_ids)
+    end
+
+    def name=(str)
+      fail TypeError unless str.is_a?(String)
+      if str.empty?
+        result = config_set('bridge_domain', 'name', @bd_ids, 'no', '')
+      else
+        result = config_set('bridge_domain', 'name', @bd_ids, '', str)
+      end
+      cli_error_check(result)
+    rescue CliError => e
+      raise "[bridge-domain #{@bd_ids}] '#{e.command}' : #{e.clierror}"
+    end
+
+    def default_name
+      sprintf('Bridge-Domain%s', @bd_ids)
+    end
+
+    # Bridge-Domain type change to fabric-control
+    def fabric_control
+      config_get('bridge_domain', 'fabric_control')
+    end
+
+    def fabric_control=(val)
+      no_cmd = (val) ? '' : 'no'
+      result = config_set('bridge_domain', 'fabric_control', @bd_ids, no_cmd)
+      cli_error_check(result)
+    rescue CliError => e
+      raise "[bridge-domain #{@bd_ids}] '#{e.command}' : #{e.clierror}"
+    end
+
+    def default_fabric_control
+      config_get_default('bridge_domain', 'fabric_control')
+    end
+
+    # Bridge-Domain member vni assigning case
+    def member_vni
+      vni_list = []
+      curr_vni = config_get('bridge_domain', 'member_vni')
+      curr_bd_vni = config_get('bridge_domain', 'member_vni_bd')
+      return '' if curr_vni.empty? || curr_bd_vni.empty?
+
+      curr_vni_list = BridgeDomain.bd_ids_to_array(curr_vni)
+      curr_bd_vni_list = BridgeDomain.bd_ids_to_array(curr_bd_vni)
+      input_bds = BridgeDomain.bd_ids_to_array(@bd_ids)
+
+      hash_map = Hash[curr_bd_vni_list.zip(curr_vni_list.map)]
+      input_bds.each do |bd|
+        vni_list << hash_map[bd]
+      end
+      return '' unless vni_list.any?
+
+      BridgeDomain.bd_list_to_string(vni_list)
+    end
+
+    def set_member_vni=(cmd, val)
+      no_cmd = (cmd) ? '' : 'no'
+      vdc = Vdc.new('default')
+      vdc.limit_resource_module_type = 'f3' unless
+        vdc.limit_resource_module_type == 'f3'
```

Comment:
These vdc commands shouldn't be here. The user needs to specify this in their cisco_vdc manifest.